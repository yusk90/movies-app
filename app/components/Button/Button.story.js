import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';

import Button from './';

const stories = storiesOf('Button', module);

stories.add('Button Component', () => (
  <div>
    <h1>Button Component</h1>
    <Button
      active={boolean('Active', false)}
    >
      Test text
    </Button>
  </div>
));
