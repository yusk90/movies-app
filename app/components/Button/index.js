import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './index.less';

const Button = ({ children, className, size }) => (
  <button
    className={classNames(
      'button',
      className,
      {
        [`button--size-${size}`]: size,
      }
    )}
  >
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  className: PropTypes.string,
  size: PropTypes.oneOf(['small']),
};

export default Button;
