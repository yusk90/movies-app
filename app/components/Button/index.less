.button {
  padding: 15px 25px;
  text-transform: uppercase;
  background-color: #b95866;
  border: none;
  border-radius: 3px;
  color: #fff;
  outline: none;
  cursor: pointer;

  &--size-small {
    padding: 5px 15px;
  }

  &--state-active {
    background-color: #000;
  }
}
