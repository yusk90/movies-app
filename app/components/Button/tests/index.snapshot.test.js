import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../index';

describe('<Button />', () => {
  it('renders button with text', () => {
    const tree = renderer
      .create(<Button>Button text</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders small button', () => {
    const tree = renderer
      .create(<Button size="small">Button text</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
