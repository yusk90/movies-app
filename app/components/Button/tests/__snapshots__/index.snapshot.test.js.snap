// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders button with text 1`] = `
<button
  className="button"
>
  Button text
</button>
`;

exports[`<Button /> renders small button 1`] = `
<button
  className="button button--size-small"
>
  Button text
</button>
`;
